src/components/elementView/elementView.tsx

```diff
@@ -1,143 +1,98 @@
-import { Classes, Spinner, Tab, Tabs } from "@blueprintjs/core";
+import { AnchorButton, Classes, Dialog } from "@blueprintjs/core";
 import "@blueprintjs/core/lib/css/blueprint.css";
 import axios from "axios";
 import * as React from "react";
-import { RouteComponentProps, withRouter } from "react-router";
+import { connect } from "react-redux";
 import { API_ROOT } from "../../api-config";
+import WrappedCreateModelForm from "../../forms/createModelForm";
+import { ElementObjectType, ElementType } from "../utils";
+import ElementTable from "./elementTable";
 import "./elementView.scss";
 
-//export interface ElementViewProps {}
-
-export class ElementView extends React.PureComponent<RouteComponentProps> {
-  public render() {
-    return (
-      <Tabs
-        className={Classes.DARK + " element-view"}
-        animate={true}
-        id="ElementViewer"
-        key={"vertical"}
-        renderActiveTabPanelOnly={false}
-        vertical={true}
-      >
-        <Tab
-          className="tab"
-          id="instance"
-          title="Instances"
-          panel={
-            <ElementTable element="instances" history={this.props.history} />
-          }
-        />
-        <Tab
-          className="tab"
-          id="model"
-          title="Models"
-          panel={<ElementTable element="models" history={this.props.history} />}
-        />
-        <Tab
-          className="tab"
-          id="rack"
-          title="Racks"
-          panel={<ElementTable element="racks" history={this.props.history} />}
-        />
-        <Tabs.Expander />
-      </Tabs>
-    );
-  }
+interface ElementViewState {
+  isOpen: boolean;
 }
-
-async function getData(path: string) {
-  console.log(API_ROOT + "api/" + path);
-  return await axios
-    //.get("https://rack-city-dev.herokuapp.com/api/" + path)
-    .get(API_ROOT + "api/" + path)
-    .then(res => {
-      const data = res.data;
-      const cols: Array<Array<string>> = data.map((item: any) => {
-        return Object.keys(item);
-      });
-      return { cols, data };
-    });
+interface ElementViewProps {
+  element: ElementType;
+  isAdmin: boolean;
 }
 
-interface ElementTableState {
-  columns: Array<string>;
-  data: any;
-}
-interface ElementTableProps {
-  element: string;
-  history: any;
-}
+function getElementData(
+  path: string,
+  token: string
+): Promise<Array<ElementObjectType>> {
```

Comment:
Creating object types allows me to directly map Axios response objects to interfaces that we defined.

Reply:
To be clear, does this just specify the return type of the function so that you can "access" data from it before the Promise is resolved?

Reply:
the return type of the function is still in a Promise, mainly because I couldn't figure out how to truly get it out of the Promise function to pass into a diff function. BUT, i basically just take this promise object and in my ElementTable I call ```    this.props.getData(this.props.type, this.props.token).then(res => {
      this.setState({
        items: res
      });
    });``` to truly get the value out of the promise.

Reply:
Also specifying that the elements inside the Promise are an array of objects of type`ElementObjectType`, which the Table needs